internal/http/services/owncloud/ocdav/ocdav_blackbox_test.go

```diff
@@ -120,13 +134,383 @@ var _ = Describe("ocdav", func() {
+						return req.Ref.Path == "/users/username"
+					})).Return(&cs3storageprovider.StatResponse{
+						Status: status.NewNotFound(ctx, "resource not found"),
+					}, nil)
+
+					handler.Handler().ServeHTTP(rr, req)
+					Expect(rr).To(HaveHTTPStatus(http.StatusMultiStatus))
+					Expect(rr).To(HaveHTTPBody("<d:multistatus xmlns:s=\"http://sabredav.org/ns\" xmlns:d=\"DAV:\" xmlns:oc=\"http://owncloud.org/ns\"><d:response><d:href>/webdav</d:href><d:propstat><d:prop><oc:name>username</oc:name><d:resourcetype></d:resourcetype><d:getcontentlength>0</d:getcontentlength><oc:favorite>0</oc:favorite></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat><d:propstat><d:prop><oc:file-parent></oc:file-parent></d:prop><d:status>HTTP/1.1 404 Not Found</d:status></d:propstat></d:response></d:multistatus>"), "Body must not be empty")
+					// TODO test listing more thoroughly
+
+				})
+			})
+		})
+
+		Describe("MKCOL", func() {
+
+			var rr *httptest.ResponseRecorder
+			var req *http.Request
+			var err error
+
+			BeforeEach(func() {
+				// setup the request
+				rr = httptest.NewRecorder()
+				req, err = http.NewRequest("MKCOL", basePath+"/newfolder", strings.NewReader(""))
+				Expect(err).ToNot(HaveOccurred())
+				req = req.WithContext(ctx)
+
+			})
+
+			When("the gateway returns OK", func() {
+				It("returns a created status", func() {
+
+					// MKCOL needs to check if the resource already exists to return the correct status
+					client.On("Stat", mock.Anything, mock.MatchedBy(func(req *cs3storageprovider.StatRequest) bool {
+						return req.Ref.Path == "/users/username/newfolder"
+					})).Return(&cs3storageprovider.StatResponse{
+						Status: status.NewNotFound(ctx, "not found"),
+					}, nil)
+
+					client.On("CreateContainer", mock.Anything, mock.MatchedBy(func(req *cs3storageprovider.CreateContainerRequest) bool {
+						return utils.ResourceEqual(req.Ref, &cs3storageprovider.Reference{
+							ResourceId: userspace.Root,
+							Path:       "./newfolder",
+						})
+					})).Return(&cs3storageprovider.CreateContainerResponse{
+						Status: status.NewOK(ctx),
+					}, nil)
+
+					handler.Handler().ServeHTTP(rr, req)
+					Expect(rr).To(HaveHTTPStatus(http.StatusCreated))
+					Expect(rr).To(HaveHTTPBody(BeEmpty()), "Body must be empty")
+					// TODO expect fileid and etag header?
+				})
+
+			})
+
+			When("the parent does not exist", func() {
+				It("returns a conflict status", func() {
+
+					// MKCOL needs to check if the resource already exists to return the correct status
+					client.On("Stat", mock.Anything, mock.MatchedBy(func(req *cs3storageprovider.StatRequest) bool {
+						return req.Ref.Path == "/users/username/newfolder"
+					})).Return(&cs3storageprovider.StatResponse{
+						Status: status.NewAborted(ctx, errors.New("parent does not exist"), ""),
+					}, nil)
+
+					handler.Handler().ServeHTTP(rr, req)
+					Expect(rr).To(HaveHTTPStatus(http.StatusPreconditionFailed))
```

Comment:
The description says "conflict", which one is correct?

Reply:
sorted out: 
- cs3 Aborted -> http precondition failed (eg mismatching lock id or etag)
- cs3 failed precondition -> http conftlict (missing intermediate dir) or method not allowed (dir already exists)